### PR TITLE
Ch intensities to table and shape.the c

### PR DESCRIPTION
**Pull request by will-moore**
In order that the Batch ROI export CSV has a ```c``` column as in my CSV screenshot at http://will-moore.github.io/presentations/2018/Users-Meeting/Workshops/Metadata-Analysis/slides/#/8
I've added this into the script. (the screenshot was after editing in Excell).

Also added a script that I used to generate Table for filtering on this Plate: http://outreach.openmicroscopy.org/webclient/?show=plate-51 (trainer-1)

To test:
 - Need to install and run the Batch_ROI_Export_to_Table script on images where the shapes have Channel set (and some shapes with no channel set). I can install script on web-dev-merge if useful.
 - Run the script on a plate to add Table. Go to http://web-dev-merge.openmicroscopy.org/webclient/?show=plate-1153 (user-3) and delete the existing bulk annotations attachment. Then run 
```
$ python maintenance/scripts/channel_minmax_to_table.py user-3 ome --server eel.openmicroscopy.org 1153
``` 
to re-create it. Test filtering by Table in parade works OK.

### CODE REVIEW

File: maintenance/scripts/channel_minmax_to_table.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+
+def run(username, password, plate_id, host, port):
+
+    conn = BlitzGateway(username, password, host=host, port=port)
+    try:
+        conn.connect()
+
+        # Create a name for the Original File
+        tablename = "Channels_Min_Mex_Intensity"
```
Comment (jburel) [inline maintenance/scripts/channel_minmax_to_table.py]: Typo ``Max`` instead of ``Mex``

File: maintenance/scripts/channel_minmax_to_table.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            if image is None:
+                continue
+            wellIds.append(well.id)
+            chCount = image.getSizeC()
+            row = []
+            print "well, image", well.id, image.id
+            for ch in image.getChannels():
+                row.append(long(ch.getWindowMin()))
+                row.append(long(ch.getWindowMax()))
+            image._re.close()
```
Comment (jburel) [inline maintenance/scripts/channel_minmax_to_table.py]: you should load the rendering def and retrieve the window min/max
Starting a rendering engine is costly
you could for example use method from the RenderingSettings Service

**Comment by will-moore** [comment]
Since I was only getting the window min/max I didn't even need the rendering engine ;)

**Comment by pwalczysko** [comment]
will you need that script on outreach ?

**Comment by will-moore** [comment]
@pwalczysko No, it's not a server-side script. It's a client script that I've already run against the Plate on outreach.
I wonder if we should add the plate import and 
```
python maintenance/scripts/channel_minmax_to_table.py user-3 ome --server eel.openmicroscopy.org 1153
```
steps to any set-up docs?
Currently the ```idr0021-data-prep.md``` covers everything we need in the workshop except for the plate set-up. It would be kinda hard for others / future-us to remember exactly what other steps were.

**Comment by jburel** [comment]
@pwalczysko This PR modifies the batch_export script so that will need to be added server side
I will tag repo after review then you can re-run playbook

**Comment by will-moore** [comment]
Ah, yes. Great, thanks.

**Comment by jburel** [comment]
```
python maintenance/scripts/channel_minmax_to_table.py user-3 ome --server eel.openmicroscopy.org 1153
well, image 303893 10994
Error while changing names: Object unloaded:object #0 (::omero::model::StatsInfo)
{
    _id = object #1 (::omero::RLong)
    {
        _val = 14410
    }
    _details = <nil>
    _loaded = False
    _version = <nil>
    _globalMin = <nil>
    _globalMax = <nil>
}

```

**Comment by will-moore** [comment]
This loads the stats, but this is not sorted by Channel:
```
            params = omero.sys.ParametersI()
            params.addId(image.getPixelsId())
            query = """select channel from Channel as channel
                       join fetch channel.statsInfo where channel.pixels.id=:id"""
            result = query_service.findAllByQuery(query, params)

            row = []
            for ch in result:
                si = ch.statsInfo
                row.extend([si.globalMin.val, si.globalMax.val])
```
Needs more work...
Really, this is taking too much time for a script that generates kinda synthetic data and already works (doesn't need to be performant).

**Comment by jburel** [comment]
The problem is that we refer to those training scripts and it is a bad practise to start an RE to the values.
You can simply retrieve the rnd settings and you will min/max values per channel

**Comment by will-moore** [comment]
The renderingdef.channelbindings don't have the min/max pixel intensities. They only have the inputstart and inputend which are rendering settings, not stats.

**Comment by jburel** [comment]
I got confused by the name of the method used

**Comment by jburel** [comment]
retrievePixDescription on IPixels will load what you want +more

**Comment by jburel** [comment]
channels_min_max

Tested with the last commit
The channels min/max are correctly red.
We could add later the channel name to the name of column.

Thanks @will-moore

**Comment by jburel** [comment]
Export script:
* ``Filter by is blank`` but the script is filtered by the channel name. Remove from display
* channel column should be renamed to something like ``Intensity for channel``
* Rename the csv file ``roi_intensities_advanced`` for something like that

**Comment by jburel** [comment]
retested
Everything works
Remove the commented out code

**Pull request merged**